### ISSUE TITLE
localePath can't fetch more than one dynamic root path

### ISSUE THREAD

**Issue by lorenaramonda** [nuxt-community/nuxt-i18n#493]
### Version

[v5.3.0](https://github.com/nuxt-i18n/releases/tag/v5.3.0)

### Reproduction link

[https://codesandbox.io/s/codesandbox-nuxt-cqsx1?fontsize=14](https://codesandbox.io/s/codesandbox-nuxt-cqsx1?fontsize=14)

### Steps to reproduce

Create an architecture like this and add a navigation link to every page created using `localePath({ name: NAMEOFTHEPAGE })`.

![Schermata 2019-10-19 alle 16.55.21.png](https://cmty.s3.eu-west-1.amazonaws.com/110467581/841c8cb0-f280-11e9-b033-859ebc22a739.png)

All of them are reachable except for _subcategory.vue in the root of the `pages` folder.

Please see codesandbox link to test it live.

### What is expected ?

_subcategory.vue should properly displaying while linking it using:
localePath({
        name: 'subcategory',
        params: { subcategory: 'subcategory' }
      })

### What is actually happening?

_subcategory.vue in the root of /pages never shows up

### Additional comments?

This properly works with a base nuxt installation without i18n activated

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt-i18n/issues/c324">#c324</a>)</em></sub></div>

**Comment by rchl** [comment]
While this module looks to be doing an incorrect thing when doing SPA navigation, that sort of structure is not really working properly, even without this module.

Here is an example without `nuxt-i18n`: https://codesandbox.io/s/codesandbox-nuxt-d50kl
Yes, it works correctly when you press the link but if you then reload page, it will go to a different route (`:category`). That's because relevant part of the generated list of routes looks like this:

```
{
    path: "/:category",
    component: _bd8ac568,
    name: "category"
  }, {
    path: "/:subcategory",
    component: _d07d95fa,
    name: "subcategory"
  }
```
and since `:category` route is first on the list, that one will handle the URL first if it matches.

So looks like a bug but I wouldn't recommend such structure anyway.

**Comment by lorenaramonda** [comment]
Oh, I see. Then how would you handle custom urls like this?

**Comment by lorenaramonda** [comment]
And also, in case I would use a structure like this: `/category/:category` would it be possible to localize `/category/` in the url? 😰

**Comment by rchl** [comment]
I don't know your requirements so it's hard for me to suggest anything but why do you want :subcategory route in the root. If you remove it, it should work fine.

As far as localizing, you can do it either through component options or module options, but not in dynamic way, I believe. Typ6u would need to hardcore that upfront. But maybe there is better way. I haven't played with that myself.

**Comment by lorenaramonda** [comment]
well, this is not the proper example, of course I won't put a :subcategory into root but maybe something like :content for static page together with :category to have seo friendly url like:
domain.com/name-of-category or
domain.com/name-of-a-static-page

so things like this are not easily reproducible?

**Comment by rchl** [comment]
Having both `:content` and `:category` is also wrong. You can't have two dynamic routes at the same level as Nuxt will not know which one to pick and will always pick the first one. Apart from the case when you programmatically navigate to the chosen one but that still leaves plain server requests not working.

**Comment by rchl** [comment]
And while I've said that this module is doing wrong thing when doing SPA navigation, technically it doesn't really.

What happens:
 - `localePath` called with an object form, resolves a path from object
 - `nuxt-link` receives a string in `to` attribute

Since `nuxt-link` receives a string, there is no way to figure out the exact route to navigate to. If it would receive an object with route name being explicitly specified then it would work but `localePath` has to resolve a string from object as it doesn't know if result will be used from `nuxt-link` or just plain `a` element.

So, to summarize, having route structure designed that way is incorrect and it's bound to have problems.
